### PR TITLE
chore(rust): add cli command to list vaults

### LINKED ISSUE #3935: Add CLI command to list vaults
Depends on: https://github.com/build-trust/ockam/issues/3933

Add a new command to list the available vaults. For each vault, show the same information as the `vault show` command.

A new `list` method must be implemented for VaultsState, similar to the NodesState's list method.

This command should prevent the user from deleting the vault if it's being used by a node. You will have to list the nodes, and check for each one its default_vault (`opts.state.nodes.list()` + `node.config.default_vault`).

---

_We love helping new contributors!_
_If you have questions or need help as you work on your first Ockam contribution, please leave a comment on [this discussion](https://github.com/ockam-network/ockam/discussions/1081)._
_If you're looking for other issues to contribute to, checkout [this discussion](https://github.com/build-trust/ockam/discussions/3206) and labels -  https://github.com/build-trust/ockam/labels/good%20first%20issue or https://github.com/build-trust/ockam/labels/help%20wanted_

### PR DESCRIPTION
**Pull request by hargut**
## Current Behavior
No `ockam vault list` available.

## Proposed Changes
Add `ockam vault list` command.

Close https://github.com/build-trust/ockam/issues/3935

## Checks

- [x] All commits in this Pull Request are [signed](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits) and Verified by Github.
- [x] All commits in this Pull Request follow the Ockam [commit message convention](https://github.com/build-trust/.github/blob/main/CONTRIBUTING.md#commit-messages).
- [x] I accept the Ockam Community [Code of Conduct](https://github.com/build-trust/.github/blob/main/CODE_OF_CONDUCT.md).
- [x] I have accepted the Ockam [Contributor License Agreement](https://github.com/build-trust/ockam-contributors/blob/main/CLA.md) by adding my Git/Github details in a row at the end of the [CONTRIBUTORS.csv](https://github.com/build-trust/ockam-contributors/blob/main/CONTRIBUTORS.csv) file in a separate pull request to the [build-trust/ockam-contributors](https://github.com/build-trust/ockam-contributors) repository.

### CODE REVIEW

File: implementations/rust/ockam/ockam_api/src/cli_state.rs [hunk 1]
```diff
@@ -171,6 +171,19 @@ impl VaultsState {
         std::os::unix::fs::symlink(original, link)?;
         self.get(name)
     }
+
+    pub fn list(&self) -> Result<Vec<VaultState>> {
+        let mut vaults: Vec<VaultState> = vec![];
+        for entry in std::fs::read_dir(&self.dir)? {
+            let entry = entry?;
+            if entry.path().extension() == Some(OsStr::from_bytes(b"json")) {
```
Comment (adrianbenavides) [inline implementations/rust/ockam/ockam_api/src/cli_state.rs]: You don't need to check the extension, just try to load the vault with `self.get(vault_name)` and if that fails, then you skip it; if you can get it, then you add it to the `vaults` vec.

File: implementations/rust/ockam/ockam_api/src/cli_state.rs [hunk 2]
```diff
@@ -171,6 +171,19 @@ impl VaultsState {
...
+                let file_name = entry.path();
+                let vault_name = file_name.file_stem().unwrap().to_str().unwrap();
```
Comment (adrianbenavides) [inline implementations/rust/ockam/ockam_api/src/cli_state.rs]: There is a utils function called `file_stem(&entry_path())` that does just that.

File: implementations/rust/ockam/ockam_command/src/vault/mod.rs [hunk 3]
```diff
@@ -41,6 +41,7 @@ pub enum VaultSubcommand {
         /// Name of the vault
         name: Option<String>,
     },
+    List {},
```
Comment (adrianbenavides) [inline implementations/rust/ockam/ockam_command/src/vault/mod.rs:61]: Would be nice to add a simple test to `commands.bats`, as part of the `"create a vault and do show on it"` test.

**Review by adrianbenavides — CHANGES_REQUESTED** [review]
Thanks for another PR @hargut. It's looking great. Left some comments with some requests 👍

File: implementations/rust/ockam/ockam_api/src/cli_state.rs (hunk 1 above)
Comment (hargut) [reply implementations/rust/ockam/ockam_api/src/cli_state.rs]: Thank you for your input.

Switched to the already implemented `list()` function which contains the suggested changes. It was introduced with the commit adding the delete part.

https://github.com/build-trust/ockam/commit/7b1d00e8b11881d6b7c422f9e7acc4c77bbc7ba5#diff-a17a5a1da1448cd1d92017a87dd5cd59c347565441dea82b61b4903d814776d7R152

File: implementations/rust/ockam/ockam_api/src/cli_state.rs (hunk 2 above)
Comment (hargut) [reply implementations/rust/ockam/ockam_api/src/cli_state.rs]: `list()` was already added, including this suggestion.

File: implementations/rust/ockam/ockam_command/src/vault/mod.rs (hunk 3 above)
Comment (hargut) [reply implementations/rust/ockam/ockam_command/src/vault/mod.rs:61]: Added the according test in `commands.bat`.

**Review by adrianbenavides — APPROVED** [review]
Looks perfect! Thank you so much @hargut 🙏

**Pull request merged**